experimental/dds/tree2/src/feature-libraries/editable-tree-2/proxies/proxies.ts

```diff
@@ -225,11 +225,22 @@ const asContextuallyTypedData = (value: ProxyNodeUnion<AllowedTypes, "javaScript
 // typed data prior to forwarding to 'LazySequence.insert*()'.
 function itemsAsContextuallyTyped(
 	iterable: Iterable<ProxyNodeUnion<AllowedTypes, "javaScript">>,
-): Iterable<ContextuallyTypedNodeData> {
-	// If the iterable is not already an array, copy it into an array to use '.map()' below.
-	const asArray = Array.isArray(iterable) ? iterable : Array.from(iterable);
+): [items: Iterable<ContextuallyTypedNodeData>, count: number] {
+	const asArray = Array.isArray(iterable)
+		? iterable.map(asContextuallyTypedData)
+		: Array.from(iterable, asContextuallyTypedData);
 
-	return asArray.map(asContextuallyTypedData);
+	return [asArray, asArray.length];
+}
+
+function* iterateRange<TTypes extends AllowedTypes>(
+	list: SharedTreeList<TTypes>,
+	start: number,
+	count: number,
+): IterableIterator<ProxyNodeUnion<TTypes>> {
+	for (let i = 0; i < count; i++) {
+		yield list[start + i];
+	}
```

Comment:
Craig has making the case that we really should following JS semantics regarding mutating during iteration.

It's a product-wide issue, so maybe we should start tagging these with an AB#xxxx as we come across them?